cognitive/src/test/scala/com/microsoft/azure/synapse/ml/cognitive/anomaly/MultivariateAnamolyDetectionSuite.scala

```diff
@@ -52,7 +54,11 @@ trait MADTestUtils extends TestBase with AnomalyKey with StorageCredentials {
   lazy val intermediateSaveDir: String =
     s"wasbs://$containerName@$storageAccount.blob.core.windows.net/intermediateData"
   lazy val fileLocation: String = DatasetUtils.madTestFile("mad_example.csv").toString
-  lazy val df: DataFrame = spark.read.format("csv").option("header", "true").load(fileLocation)
+  lazy val fileSchema: StructType = StructType(Array(
+    StructField(timestampColumn, StringType, nullable = true)
+  ) ++ inputColumns.map(inputCol => StructField(inputCol, DoubleType, nullable = true)))
+  lazy val df: DataFrame = spark.read.format("csv")
+    .option("header", "true").schema(fileSchema).load(fileLocation)
```

Comment:
Does this not infer proper schema here?

Reply:
Yes... I don't know why but it seems fails to infer the schema :(